src/elements/components/TextNodes.jsx

```diff
@@ -31,67 +46,79 @@ function TextNodes({
-        let text = op.insert;
-        if (values) {
-          // replace placeholder variables and populate newlines
-          text = op.insert.replace(TEXT_VARIABLE_PATTERN, (pattern) => {
-            const pStr = pattern.slice(2, -2);
-            if (pStr in values) {
-              const pVal = values[pStr];
-              if (Array.isArray(pVal)) {
-                if (pVal.length === 0) {
-                  return pattern;
-                } else if (
-                  isNaN(element.repeat) ||
-                  element.repeat >= pVal.length
-                ) {
-                  return stringifyWithNull(pVal[0]);
-                } else {
-                  return stringifyWithNull(pVal[element.repeat]);
-                }
-              } else return stringifyWithNull(pVal);
-            } else return pattern;
-          });
-        }
+    const spanProps = editable
+      ? editableProps(onTextSelect, onTextInput, onTextBlur)
+      : {};
+    return (
+      <span {...spanProps} key={text}>
+        {delta
+          .filter((op) => op.insert)
+          .map((op, i) => {
+            let text = op.insert;
+            if (values) {
+              // replace placeholder variables and populate newlines
+              text = op.insert.replace(TEXT_VARIABLE_PATTERN, (pattern) => {
+                const pStr = pattern.slice(2, -2);
+                if (pStr in values) {
+                  const pVal = values[pStr];
+                  if (Array.isArray(pVal)) {
+                    if (pVal.length === 0) {
+                      return pattern;
+                    } else if (
+                      isNaN(element.repeat) ||
+                      element.repeat >= pVal.length
+                    ) {
+                      return stringifyWithNull(pVal[0]);
+                    } else {
+                      return stringifyWithNull(pVal[element.repeat]);
+                    }
+                  } else return stringifyWithNull(pVal);
+                } else return pattern;
+              });
+            }
 
-        let onClick = () => {};
-        let cursor = 'inherit';
-        const attrs = op.attributes || {};
-        if (attrs.font_link) {
-          onClick = () => window.open(attrs.font_link, '_blank');
-          cursor = 'pointer';
-        } else if (isNum(attrs.start) && isNum(attrs.end)) {
-          onClick = () => {
-            handleRedirect({
-              metadata: {
-                elementType: 'text',
-                elementIDs: [element.id],
-                trigger: 'click',
-                start: attrs.fullSpan ? undefined : attrs.start,
-                end: attrs.fullSpan ? undefined : attrs.end
+            const attrs = op.attributes || {};
+            let onClick = () => {};
+            let cursor = 'text';
+            if (!editable) {
+              cursor = 'inherit';
+              if (attrs.font_link) {
+                onClick = () => window.open(attrs.font_link, '_blank');
```

Comment:
this isn't needed on modern browser versions, but still probably a good idea to do `window.open(attrs.font_link, '_blank', 'noopener noreferrer')` https://mathiasbynens.github.io/rel-noopener